assets/app/scripts/services/applicationGenerator.js

```diff
@@ -202,6 +202,13 @@ angular.module("openshiftConsole")
           type: "ImageChange"
         });
       }
+
+      // User can input a URL that contains a ref
+      var urlAndRef = input.buildConfig.sourceUrl.split("#");
```

Comment:
can we just parse with URI to get the actual fragment? should take care of the "fragment containing #" case